### PR TITLE
fix: Table and DataGrid should not remove cells from the accessibility tree

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

The header cell for the radio selection column was removed from the a11y tree via two separate mechanisms:
1. The HTML `hidden` attribute was applied to the cell root, because of the Fluent `hidden` prop of the same name
2. A `visibility: hidden` style was applied based on the hidden state

Removing a cell from a table breaks the table structure, as well as screen reader table navigation, which resulted in a high-severity a11y bug.

## New Behavior

While we could have made the Fluent `hidden` prop not apply that HTML `hidden` prop, that would also have meant that authors who intended to add the HTML `hidden` prop (e.g. to hide an entire column, and not just the header) would have been unable to.

Instead, I deprecated `hidden` and added an `invisible` prop to do the same thing.

## Related Issue(s)

- Fixes [ADO issue](https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/18749)

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Avatar-mount_base.html">632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Avatar-mount.html">635</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Button-mount_base.html">303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Button-mount.html">318</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Field-mount_base.html">1143</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Field-mount.html">1132</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProvider-mount_base.html">704</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProvider-mount.html">698</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProviderWithTheme-mount_base.html">85</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">33</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">34</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/MakeStyles-mount_base.html">860</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/MakeStyles-mount.html">866</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Persona-mount_base.html">1771</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/Persona-mount.html">1723</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/SpinButton-mount_base.html">1349</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31068/perf-test-react-components/SpinButton-mount.html">1350</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr> | `168.498 kB`<br />`46.732 kB` | `168.508 kB`<br />`46.729 kB` | `10 B` <br />`-3 B` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr> | `137.918 kB`<br />`37.149 kB` | `137.912 kB`<br />`37.14 kB` | `-6 B` <br />`-9 B` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> | `76.334 kB`<br />`20.553 kB` | `76.328 kB`<br />`20.545 kB` | `-6 B` <br />`-8 B` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr> | `74.977 kB`<br />`20.155 kB` | `74.971 kB`<br />`20.148 kB` | `-6 B` <br />`-7 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `71.104 kB`<br />`20.52 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `221.141 kB`<br />`62.432 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `43.591 kB`<br />`14.356 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `7.944 kB`<br />`2.588 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr> | `45.324 kB`<br />`14.116 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/dd21ca3927cfff8be57336f213d4556333dc038d'>dd21ca3927cfff8be57336f213d4556333dc038d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by ling1726 — APPROVED** [review]

File: packages/react-components/react-table/stories/Table/SingleSelect.stories.tsx [hunk 1]
```diff
@@ -143,7 +143,7 @@ export const SingleSelect = () => {
     <Table aria-label="Table with single selection">
       <TableHeader>
         <TableRow>
-          <TableSelectionCell type="radio" hidden />
+          <TableSelectionCell type="radio" invisible hidden />
```
Comment (smhigley) [inline packages/react-components/react-table/stories/Table/SingleSelect.stories.tsx:146]:
```suggestion
          <TableSelectionCell type="radio" invisible />
```

oops forgot to remove hidden

**Pull request merged**